### DISCUSSION TITLE
Homer 10 vs Homer 7

### DISCUSSION THREAD

**Discussion by elianero** [sipcapture/homer#613] [General]
Hello,

Is it possible to know what are the key differences between Homer 10 and Homer 7?
So far i found just:
- No SQL Databases. Powered by stock Observability APIs such as [qryn](https://qryn.dev/)
- More than Packets and PCAPs. Ingest and process Logs, Metrics and Traces.

Thank you!

Cheers!

**Comment by lmangani** [comment]
Hello @elianero thanks for the question. The current key differences are those you listed, and those involve a lot of changes already. HOMER 10 is a design preview and the final results will greatly depend on user feedback, so feel free to try it out and provide us with your thoughts 👍 as per always there's going to be some pain involved with achieving a longer-term vision for the stack.

**Comment by mkpaz** [comment]
I should say it's very confusing that while qryn states it ingests everything, its official doc never mentions single and the most important thing. It's HEP. In fact, searching HEP in both qryn doc and repository returns nothing. There's also nor qryn nor clickhouse support from the heplify server side. All I see is about logs, logs, logs, metrics, metrics, telemetry. These also a presentation from the OpenSIPS conference [here](https://www.opensips.org/events/Summit-2022Athens/assets/presentations/OpenSIPS_Summit2022_Lorenzo_Mangani_HOMER%202022%20UPDATE.pdf) and it says nothing about call tracing and call flow diagrams, which is one of the killer features. 

Sorry to ask this, but is Homer stack is still about VoIP monitoring and will be maintained in the foreseeable future? I'm just worried that it will become jack of all trades. Does everything, good at nothing. Another concern is that using NodeJS on the backend side is well ... questionable from the performance perspective (not to mention its the single entry point). Sorry, I ask too many questions. But is qryn a battle-tested solution or it's still a proof-of-concept/MVP?

**Comment by lmangani** [comment]
@mkpaz I understand your confusion, this always comes with breaking changes. All the HEP data is stored in qryn as JSON objects and all the HOMER metrics are stored in qryn as prometheus metrics. It's that simple. HOMER 10 is just in experimental/preview mode for the brave ones willing to try and give us feedback, so no panic required. qryn does not replace homer, they just work together. We develop both applications. 

The most important point: Homer is 100% about VoIP and remains focused on VoIP and WebRTC, that's our religion, we just expand its horizons and want the data it produces to be harmonized with logs, metrics and traces from other systems running the VoIP applications.

**Reply by spady7** [reply]
> .....All the HEP data is stored in qryn as JSON objects and all the HOMER metrics are stored in qryn as prometheus metrics. It's that simple. 

@lmangani Having a look to entire project I am bit confused on what you wrote. Does data stored in qryn or clickhouse? From what I've understood qryn send data to clickhouse and there there are real data (stored in colums instead of raw ...etc.).
Am I wrong?

**Reply by Dletta** [reply]
qryn is the collector for writes and reads. Clickhouse is where the data is written to and read from.

**Comment by Pavan0520** [comment]
Hi. I installed homer10 with docker. I am new to Homer10. How can I confirm my server to Homer? Can somebody assist me with correctly installing and configuring my server?

**Comment by spady7** [comment]
Another point that is not clear to me is that in homer 7 i was able to select multiple messages/transactions to view entire session (for instance). Is this still possibile with homer 10? i tried in grafana dashboard but I didn't find a way.
Regards

**Reply by lmangani** [reply]
I assume you're referring to correlation. Right now there's no "selection" in Grafana Flow yet but we're working on it. Please open a proposal issue and we'll make sure it's counted in.

**Comment by adubovikov** [comment]
Hi all,

This thread is from the Homer 10 preview era. A short **2026 update** for anyone landing here today.

### Where things stand now

| | Homer 7 | Homer 10 (preview) | **Homer 11 (current)** |
|---|---------|-------------------|------------------------|
| Status | stable legacy stack | experimental / design preview | **released** ([releases](https://github.com/sipcapture/homer/releases)) |
| Storage | PostgreSQL/Timescale (`hep_proto_*`) | qryn (preview stack) | **DuckLake** (Parquet + object storage) |
| UI | homer-ui (Angular) | mostly Grafana-centric preview | **bundled SPA** (search, call flow, dashboards) |
| Ingest | HEPv3 (heplify, etc.) | HEP + logs/metrics/traces | **HEPv3** + OTLP + Line Protocol |
| Config DB | PostgreSQL `homer_config` | preview-specific | **DuckDB** (coordinator settings) |

**Homer 10 was never a full production replacement for Homer 7.** It was a preview to explore observability APIs (qryn, Grafana-first UX). Development continued into **Homer 11**, which is the monolith we ship today on branch `homer11`.

### Answers to open questions in this thread

**[@mkpaz](https://github.com/mkpaz)** — VoIP/SIP capture is still the core. Homer 11 ingests the same HEP agents and adds SIP search, correlation, call flow, PCAP export, QoS/RTCP. Logs/metrics/traces are additive, not a replacement for SIP tracing.

**[@spady7](https://github.com/spady7)** — multi-message / session correlation and call flow are in the **Homer 11 UI** (transaction view, correlation scripts). You do not need Grafana-only flow for the primary SIP trace UX anymore.

**[@Pavan0520](https://github.com/Pavan0520)** — for a new install today, use **Homer 11**, not Homer 10 preview docker. Quick start: [releases](https://github.com/sipcapture/homer/releases) or `examples/docker` in this repo. Point heplify/hepagent at the HEP ingest port (default UDP 9060).

**qryn ([@spady7](https://github.com/spady7))** — that was the H10 preview path (qryn as the observability ingest/query layer). **Homer 11 does not use qryn.** Storage is DuckLake; query via coordinator REST API, `homer search` CLI, optional Grafana FlightSQL.

### Migrating from Homer 7

Experimental migration docs:

- `docs/untested/MIGRATION_HOMER7.md` (settings + HEP replay)

Homer 7 remains usable for existing deployments. New deployments should plan for **H11**.

### Bottom line

What you listed in the opening post (no SQL DB for capture, observability APIs, logs/metrics/traces) describes the **H10 direction**. What shipped is **H11**: VoIP-first, all-in-one homer-core, DuckLake backend, full web UI, backwards compatible HEP ingest.

Happy to answer H7 → H11 migration questions in a fresh thread if useful.
